foursquaresearch/views.py

```diff
@@ -73,26 +55,23 @@ def remove_from_favorites(request):
     if request.method == "POST":
         name = request.POST['name']
         location = request.POST['location']
+        obj = Favorite.objects.filter(user=request.user, place=Place.objects.filter(name=name, location=location))
         Favorite.objects.filter(user=request.user, place=Place.objects.filter(name=name, location=location)).delete()
         return redirect('favorites')
 
 def favorites(request):
-    history = PreviousSearch.objects.order_by('-created_date')
     logged_in = True
-    username = request.user.username
-
+    user = request.user
+    history = get_history(logged_in, user)
     favs = Favorite.objects.filter(user=request.user)
```

Comment:
in python we almost never use nicknames or shortened variable names. 
instead of favs, favourites is more readable approach. It is harder to write for sure, but we read 10x times more than we write ;)